polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -846,10 +847,51 @@ private void validateLocationForTableLike(
       TableIdentifier identifier,
       String location,
       PolarisResolvedPathWrapper resolvedStorageEntity) {
-    Optional<PolarisStorageConfigurationInfo> optStorageConfiguration =
-        PolarisStorageConfigurationInfo.forEntityPath(
-            callContext.getPolarisCallContext().getDiagServices(),
-            resolvedStorageEntity.getRawFullPath());
+
+    boolean allowTableLocationOutsideNamespaceLocation =
+        callContext
+            .getPolarisCallContext()
+            .getConfigurationStore()
+            .getConfiguration(
+                callContext.getPolarisCallContext(),
+                PolarisConfiguration.ALLOW_TABLE_LOCATION_OUTSIDE_NAMESPACE_LOCATION);
```

Comment:
This seems redundant with `ALLOW_UNSTRUCTURED_TABLE_LOCATION`. Why introduce a new configuration option? Can we just keep using the other one?

Reply:
That's true, maybe we can stick with `ALLOW_UNSTRUCTURED_TABLE_LOCATION` -- but that currently doesn't allow escape outside the catalog, so it doesn't do what it seems to say on the tin. I thought something more explicit might be preferable.